src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -1313,9 +1313,9 @@ public void visitConditional(JCConditional tree) {
                 @Override
                 public void visitReference(JCMemberReference tree) {
                     if (sRet.hasTag(VOID)) {
-                        result &= true;
+                        result = true;
```

Comment:
Isn't the equivalent statement to
result &= true
just
result
?

Reply:
Agreeing with jddarcy:
`a & true === a`
so current code just keeps the value of `result` while the new one sets `result` to `true`.